spec/models/azure_engine_spec.rb

```diff
@@ -0,0 +1,153 @@
+    end
+  end
+
+  describe 'queue_endpoint' do
+    it 'return URL that specifies queue endpoint' do
+      expect(AzureEngine.queue_endpoint).to eq('https://azure_storage_account_name.queue.core.windows.net/')
+    end
+  end
+
+  describe 'Blob Queue' do
+    it 'passes all tests' do
+      module Azure
+        module Queue
+          class DummyQueueService
+            def create_message(queue_name, message)
+              nil
+            end
+            def list_messages(queue_name, timeout, options = {})
+              []
+            end
+            def create_queue(queue_name)
+              nil
+            end
+            def delete_message(queue_name, message_object_id, message_object_pop_receipt)
+              nil
+            end
+          end
+        end
+        module Entity
+          module Queue
+            class DummyMessage
+              def id
+                1
+              end
+              def pop_receipt
+                ''
+              end
+            end
+          end
+        end
+      end
+      allow_any_instance_of(Azure::ClientServices).to receive(:queues).and_return(Azure::Queue::DummyQueueService.new)
+      expect(AzureEngine.send_message('hoge')).to eq(nil)
+      expect(AzureEngine.receive_message(10)).to eq([])
+      expect(AzureEngine.message_exist?([])).to eq(false)
+      expect(AzureEngine.message_exist?([Azure::Entity::Queue::DummyMessage.new])).to eq(true)
+      expect(AzureEngine.delete_message(Azure::Entity::Queue::DummyMessage.new)).to eq(nil)
+      expect(AzureEngine.batch_delete([Azure::Entity::Queue::DummyMessage.new]).class.name).to eq("Array")
+    end
+  end
+
+  describe 'Blob Storage' do
+    it 'passes all tests' do
+      module Azure
+        module Blob
+          class DummyBlobObject
+            def name
+              'dummy'
+            end
+          end
+          class DummyBlobService
+            def create_block_blob(container, key, content)
+              nil
+            end
+            def list_blobs(container, options = {})
+              result = []
+              result.push(Azure::Blob::DummyBlobObject.new)
+              result
+            end
+            def get_blob(container, key)
+              return nil, 'hoge'
+            end
+            def delete_blob(container, key)
+              nil
+            end
+            def generate_uri(uri, options = {})
+              'http://www.example.com'
+            end
+          end
+        end
+        module Contrib
+          module Auth
+            class DummySharedAccessSignature
+              def sign
+                'https://signed.example.com'
+              end
+            end
+          end
+        end
+      end
+
+      allow(Azure::Blob::BlobService).to receive(:new).and_return(Azure::Blob::DummyBlobService.new)
+      files = []
+      Tempfile.create("foo") do |f|
```

Comment:
__[rubocop]__
* Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)